File: C/BEGINNER/1181 Line in Array.c
#include <stdio.h>
int main()
{
    int operationLine, i, j;
    char operation[2];
    double matrix[12][12], sum;
    sum = 0;
    scanf("%d %s", &operationLine, operation);
    for (i = 0; i < 12; i++)
        for (j = 0; j < 12; j++)
            scanf("%lf", &matrix[i][j]);
    for (i = 0; i < 12; i++)
        for (j = 0; j < 12; j++)
            if (i == operationLine)
                sum += matrix[i][j];
    if (operation[0] == 'S')
        printf("%.1lf\n", sum);
    else
        printf("%.1lf\n", (sum / 12.0));
    return 0;
}

File: Python/BEGINNER/1181 Line in Array.py
# SUBPROGRAM
def create_array():
	my_array = []
	for col in range(12):
		row = []
		for lin in range(12):
			row.append(float(input()))
		my_array.append(row)
	return my_array


def sum_a_row(op_row, my_array):
	row_sum = 0
	for col in range(len(my_array)):
		row_sum += my_array[op_row][col]
	return row_sum


def average_a_row(op_row, my_array):
	row_sum = sum_a_row(op_row, my_array)
	row_average = row_sum / len(my_array[op_row])
	return row_average


# MAIN
operation_row = int(input())
operation = input()
array = create_array()
if operation == 'S':
	sum = sum_a_row(operation_row, array)
	print('%.1f' % sum)
else:
	average = average_a_row(operation_row, array)
	print('%.1f' % average)
